lib/src/widgets/yaru_search_field.dart

```diff
@@ -0,0 +1,351 @@
+            errorBorder: border,
+            focusedBorder: border,
+            contentPadding: widget.contentPadding,
+            hintText: widget.hintText,
+            fillColor: widget.fillColor ?? theme.dividerColor,
+            hoverColor:
+                (widget.fillColor ?? theme.dividerColor).scale(lightness: 0.1),
+            suffixIconConstraints:
+                const BoxConstraints(maxWidth: kYaruTitleBarItemHeight),
+            suffixIcon: widget.onClear == null
+                ? null
+                : IconButton(
+                    style: IconButton.styleFrom(
+                      shape: RoundedRectangleBorder(
+                        borderRadius: suffixRadius,
+                      ),
+                    ),
+                    onPressed: _clear,
+                    icon: ClipRRect(
+                      borderRadius: suffixRadius,
+                      child: const Icon(
+                        YaruIcons.edit_clear,
+                      ),
+                    ),
+                  ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  void _clear() {
+    widget.onClear?.call();
+    _controller.clear();
+  }
+}
+
+/// Combines [YaruSearchField], [YaruSearchButton] and any title [Widget] in a [Stack]
+class YaruSearchTitleField extends StatefulWidget {
+  const YaruSearchTitleField({
+    super.key,
+    required this.searchActive,
+    required this.title,
+    this.width = 190,
+    this.titlePadding = const EdgeInsets.only(left: 45.0),
+    this.autoFocus = true,
+    this.text,
+    this.hintText,
+    this.onSubmitted,
+    this.onClear,
+    this.onSearchActive,
+    this.onChanged,
+    this.alignment = Alignment.centerLeft,
+    this.radius = const Radius.circular(kYaruTitleBarItemHeight),
+    this.style = YaruSearchFieldStyle.filled,
+  });
+
+  final bool searchActive;
+  final Widget title;
+  final double width;
+  final EdgeInsets titlePadding;
+  final bool autoFocus;
+  final String? text;
+  final String? hintText;
+  final void Function(String? value)? onSubmitted;
+  final void Function(String)? onChanged;
+  final void Function()? onClear;
+  final void Function()? onSearchActive;
+  final Alignment alignment;
+  final Radius radius;
+  final YaruSearchFieldStyle style;
+
+  @override
+  State<YaruSearchTitleField> createState() => _YaruSearchTitleFieldState();
+}
+
+class _YaruSearchTitleFieldState extends State<YaruSearchTitleField> {
+  late bool _searchActive;
+
+  @override
+  void initState() {
+    super.initState();
+    _searchActive = widget.searchActive;
+  }
+
+  @override
+  void didUpdateWidget(covariant YaruSearchTitleField oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _searchActive = widget.searchActive;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: widget.width,
+      child: ClipRRect(
+        borderRadius: BorderRadius.all(widget.radius),
+        clipBehavior: Clip.antiAliasWithSaveLayer,
```

Comment:
why `Clip.antiAliasWithSaveLayer`? as the docs say, it's much slower than `Clip.antiAlias` and should rarely be used.